proposals/bound-callable-references.md

```diff
@@ -27,10 +27,50 @@ Support "bound callable references" and "bound class literals".
 - Drop the "callable reference to object/companion member" error. Leave other diagnostics for now.
 - Support the following syntax: `<expression>::class` (see semantics below).
 
+### Parsing
+
+Now the LHS of a callable reference expression or a class literal expression (or *double colon expressions*) can mean either an expression or a type, and it's impossible to figure out what it is by looking only at the AST. So we always parse an expression, with optional question marks after it to support the case of nullable types.
+
+```
+callableReferenceExpression
+    : (expression "?"*)? "::" SimpleName typeArguments?
+    ;
```

Comment:
No, we're parsing it as a slightly special-cased postfix expression. This rule is intentionally simplified here for better readability. Also I'm not sure what productions to include here to explain precisely what's happening...